pubsub.go

```diff
@@ -0,0 +1,125 @@
+package lazyhttp
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/dendhi31/lazyhttp/redismaint"
+)
+
+func (httprequest *Client) Consumer() error {
+	config := redismaint.Configuration{
+		RedisURL:   httprequest.PubSubServer,
+		ContexName: "first",
+		Logger:     httprequest.Logger,
+		Handler:    httprequest.optimisticReq,
+	}
+
+	rmaint, err := redismaint.New(config)
+	if err != nil {
+		return err
+	}
+
+	term := make(chan os.Signal, 1)
+	signal.Notify(term, os.Interrupt, syscall.SIGTERM)
+
+	go func(r *redismaint.Consumer) {
+		log.Println("lazyhttp consumer started")
+		r.Run()
+	}(rmaint)
+	//send sample schedule
+	select {
+	case <-rmaint.Err():
+		return err
+	case <-term:
+		log.Println("lazyhttp consumer stopped")
+		rmaint.Stop()
+		return nil
+	}
+}
+
+func (httprequest *Client) optimisticReq(ctx context.Context, url string, action string, payload []byte, header map[string]string, key string) (int, []byte, error) {
+	var responseBody []byte
+	var err error
+	var code int
+
+	mCtx, cancel := context.WithTimeout(context.Background(), httprequest.WaitHttp*time.Second)
+	defer cancel()
+
+	redisChan := make(chan redisChannel, 1)
+	httpChan := make(chan httpChannel, 1)
+
+	go func() {
```

Comment:
you need to pass those object below by reference 
(context, key, httprequest) properly

Reply:
do we have context timeout operation to limit the timeout while obtaining data from redis? (optinal)